test/db2.discover.test.js

```diff
@@ -15,10 +15,14 @@ require('./init.js');
 // var DataSource = require('loopback-datasource-juggler').DataSource;
 var db, config;
 
-before(function() {
-  db = global.getDataSource();
-  config = global.config;
-});
+// Note: remove this before() function because it is causing segfault in CI.
+// All the tests in this file are skipped to begin with, so there is no use
+// to call this function anyway.
+// Make sure this will be uncommented when enabling the tests.
+// before(function() {
```

Comment:
@rmg helped to investigate and opened https://github.com/ibmdb/node-ibm_db/issues/259.  I believe by calling `getDataSource()`, it causes the connection to be initiated which caused segmentation fault.  
The other possibility, as Ryan suggested, is to get the data source info inside the describe. But given the tests are skipped anyway, i'm not sure if it makes sense to make this change.